test/e2e/networking_perf.go

```diff
@@ -0,0 +1,249 @@
+			TryEachPod(f.Client, f.Namespace.Name, "app", "iperf-e2e-cli-pod",
+				func(p api.Pod) {
+					resultS, err := lookForStringInLog(f.Namespace.Name, p.Name, "iperf-client", "0-10.0", 1*time.Second)
+					if err == nil {
+						Logf(resultS)
+						ipr := NewIPerf(resultS)
+						bandwidthMap[p.Name+"-"+ipr.cli+"-"+ipr.id+"--server="+ipr.server] = ipr.bandwidthBits
+						if len(bandwidthMap) >= expectedCli {
+							Logf("Got enough results (%v / %v) to terminate the test...", len(bandwidthMap), expectedCli)
+						} else {
+							Logf("Only got results from %v out of %v...", len(bandwidthMap), expectedCli)
+						}
+					} else {
+						Logf("No result for %v. ", p.Name)
+					}
+				})
+			time.Sleep(1 * time.Second)
+		}
+
+		fmt.Println("[begin] Node,Bandwith CSV")
+		for node, bandwidth := range bandwidthMap {
+			fmt.Println(node, "\t ", fmt.Sprintf("%E", float64(bandwidth)))
+		}
+		fmt.Println("[end] Node,Bandwith CSV")
+
+		for node, bandwidth := range bandwidthMap {
+			Logf("%v had bandwidth %v.  Ratio to expected (%v) was %f", node, bandwidth, expectedBandwidth, float64(bandwidth)/float64(expectedBandwidth))
+		}
+	})
+}
+
+// A struct modelling an iperf record.  The records look something like this.
+// The struct fields are in order !
+// 20160314154239,172.17.0.3,34152,172.17.0.2,5001,3,0.0-10.0,33843707904,27074774092
+type IPerfResult struct {
+	date          string // field 1 in the csv
+	cli           string // field 2 in the csv
+	cliPort       int64  // ...
+	server        string
+	servPort      int64
+	id            string
+	interval      string
+	transferBits  int64
+	bandwidthBits int64
+}
+
+// parse parses an IPerf CSV output line into an IPerfResult
+func NewIPerf(csvLine string) IPerfResult {
+	slice := StrSlice(strings.Split(csvLine, ","))
+	if len(slice) != 9 {
+		Failf("Incorrect fields in the output: %v (%v out of 9)", slice, len(slice))
+	}
+	i := IPerfResult{}
+	i.date = slice.get(0)
+	i.cli = slice.get(1)
+	i.cliPort = intOrFail("client port", slice.get(2))
+	i.server = slice.get(3)
+	i.servPort = intOrFail("server port", slice.get(4))
+	i.id = slice.get(5)
+	i.interval = slice.get(6)
+	i.transferBits = intOrFail("transfer port", slice.get(7))
+	i.bandwidthBits = intOrFail("bandwidth port", slice.get(8))
+	return i
+}
+
+// Utility functions for parsing data from iperf.
+
+type StrSlice []string
+
+func (s StrSlice) get(i int) string {
+	if i >= 0 && i < len(s) {
+		return s[i]
+	}
+	return ""
+}
+
+// return an int or fail.  the first arg is for easy debugging purposes only in case of failure.
+func intOrFail(debugName string, rawValue string) int64 {
+	value, err := strconv.ParseInt(rawValue, 10, 64)
+	if err != nil {
+		// Failf("Failed parsing value %v from the string '%v' as an integer", debugName, rawValue)
+	}
+	return value
+}
+
+// This is an opinionated utility method, but could be used with any test that wants to spin N pods on a node.
```

Comment:
I think this would be nicer as an iperf runner. Something that encapsulates all the logic of running iperf that I can invoke from a non-e2e go program as well. It should take just the params for running iperf as struct members, the svc name/ip whatever other runtime args as method args, and look something like:
```
type IPerfRunner struct {
 // iperf args
}
func (iperf *IPerfRunner) Run(port, svcName) IPerfResults {}
```

Then anyone can extend this bench, or add your iperf stuff to other tests.